pkg/materialize/source_table_kafka.go

```diff
@@ -0,0 +1,389 @@
+	exposeProgress   IdentifierSchemaStruct
+}
+
+func NewSourceTableKafkaBuilder(conn *sqlx.DB, obj MaterializeObject) *SourceTableKafkaBuilder {
+	return &SourceTableKafkaBuilder{
+		SourceTableBuilder: NewSourceTableBuilder(conn, obj),
+	}
+}
+
+func (b *SourceTableKafkaBuilder) IncludeKey() *SourceTableKafkaBuilder {
+	b.includeKey = true
+	return b
+}
+
+func (b *SourceTableKafkaBuilder) IncludeHeaders() *SourceTableKafkaBuilder {
+	b.includeHeaders = true
+	return b
+}
+
+func (b *SourceTableKafkaBuilder) IncludePartition() *SourceTableKafkaBuilder {
+	b.includePartition = true
+	return b
+}
+
+func (b *SourceTableKafkaBuilder) IncludeOffset() *SourceTableKafkaBuilder {
+	b.includeOffset = true
+	return b
+}
+
+func (b *SourceTableKafkaBuilder) IncludeTimestamp() *SourceTableKafkaBuilder {
+	b.includeTimestamp = true
+	return b
+}
+
+func (b *SourceTableKafkaBuilder) IncludeKeyAlias(alias string) *SourceTableKafkaBuilder {
+	b.includeKey = true
+	b.keyAlias = alias
+	return b
+}
+
+func (b *SourceTableKafkaBuilder) IncludeHeadersAlias(alias string) *SourceTableKafkaBuilder {
+	b.includeHeaders = true
+	b.headersAlias = alias
+	return b
+}
+
+func (b *SourceTableKafkaBuilder) IncludePartitionAlias(alias string) *SourceTableKafkaBuilder {
+	b.includePartition = true
+	b.partitionAlias = alias
+	return b
+}
+
+func (b *SourceTableKafkaBuilder) IncludeOffsetAlias(alias string) *SourceTableKafkaBuilder {
+	b.includeOffset = true
+	b.offsetAlias = alias
+	return b
+}
+
+func (b *SourceTableKafkaBuilder) IncludeTimestampAlias(alias string) *SourceTableKafkaBuilder {
+	b.includeTimestamp = true
+	b.timestampAlias = alias
+	return b
+}
+
+func (b *SourceTableKafkaBuilder) Format(f SourceFormatSpecStruct) *SourceTableKafkaBuilder {
+	b.format = f
+	return b
+}
+
+func (b *SourceTableKafkaBuilder) Envelope(e KafkaSourceEnvelopeStruct) *SourceTableKafkaBuilder {
+	b.envelope = e
+	return b
+}
+
+func (b *SourceTableKafkaBuilder) KeyFormat(k SourceFormatSpecStruct) *SourceTableKafkaBuilder {
+	b.keyFormat = k
+	return b
+}
+
+func (b *SourceTableKafkaBuilder) ValueFormat(v SourceFormatSpecStruct) *SourceTableKafkaBuilder {
+	b.valueFormat = v
+	return b
+}
+
+func (b *SourceTableKafkaBuilder) ExposeProgress(e IdentifierSchemaStruct) *SourceTableKafkaBuilder {
+	b.exposeProgress = e
+	return b
+}
+
+func (b *SourceTableKafkaBuilder) Create() error {
+	return b.BaseCreate("kafka", func() string {
+		q := strings.Builder{}
+		var options []string
+
+		// Format
+		if b.format.Avro != nil {
+			if b.format.Avro.SchemaRegistryConnection.Name != "" {
+				options = append(options, fmt.Sprintf(`FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION %s`, QualifiedName(b.format.Avro.SchemaRegistryConnection.DatabaseName, b.format.Avro.SchemaRegistryConnection.SchemaName, b.format.Avro.SchemaRegistryConnection.Name)))
+			}
+			if b.format.Avro.KeyStrategy != "" {
+				options = append(options, fmt.Sprintf(`KEY STRATEGY %s`, b.format.Avro.KeyStrategy))
```

Comment:
Here we would not need to escape this is a keyword. If we were to escape it it would result in the following error on the Materialize side:

```
Expected one of ID or LATEST or INLINE, found string literal "INLINE"
```

We do however have validation in place on the terraform schema side already, so a user will only be able to specify one of the following values: https://github.com/MaterializeInc/terraform-provider-materialize/blob/68319193d1576420971d5e4cb95b60e8ab13fab3/pkg/resources/config.go#L58-L62